docs(*): added applications metrics through kuma docs (#827)
kumahq/kuma-website: docs/docs/dev/policies/traffic-metrics.md

::::

## Expose metrics from applications
 
In addition to exposing metrics from the data plane proxies, you might want to expose metrics from applications running next to the proxies. You can do this in 2 different ways:
 
* directly from the applications,
* through Kuma sidecar.
 

> ### Expose metrics directly from the applications

Reviewer: Do we really want to explain this? That has little to do with kuma.

Author: It's more about configuration, we don't really explain it too deeply. It's the same thing that was explained previously.

Reviewer: I think it's just contributing to information overload. Let's remove this first option completely.

To consume paths protected by mTLS, you need Traffic Permission that lets Prometheus consume applications.

### Expose metrics through the Kuma sidecar.

> Kuma allows scraping Prometheus metrics from the applications endpoint running in the same `Pod` or `VM`. Later those metrics are aggregated and exposed at the same `port/path` as Dataplane metrics. It is possible to configure it at the `Mesh` level, for all the applications in the `Mesh`, or just for specific applications. It's worth mentioning that any configuration change requires redeployment.

Reviewer: ```suggestion
Kuma allows scraping Prometheus metrics from the applications endpoint running in the same `Pod` or `VM`. Later those metrics are aggregated and exposed at the same `port/path` as Dataplane metrics. It is possible to configure it at the `Mesh` level, for all the applications in the `Mesh`, or just for specific applications.
This is especially useful when mTLS is enabled and the prometheus scraper doesn't use mTLS.

::: tip
Any configuration change requires redeployment of the dataplane.
:::
```

      aggregate:
        my-service: # name of the metric, required to later disable/override at dataplane configuration
          path: "/metrics/prometheus"
          port: 8888
        other-sidecar:
          # default path is going to be used, default: /metrics
          port: 8000
```
:::
::::

This configuration will cause every application in the `Mesh` to be scrapped for metrics by the Kuma dataplane. If you need to expose metrics only for the specific application it is possible through `annotation` for Kubernetes or `Dataplane` configuration for Universal deployment.

:::: tabs :options="{ useUrlFragment: false }"
::: tab "Kubernetes"
Kubernetes allows to configure it through annotations. In case to configure you can use `prometheus.metrics.kuma.io/aggregate-<name>-(path/port/enabled)`, where name is used to match the `Mesh` configuration and override or disable it.
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
 namespace: kuma-example
 name: kuma-tcp-echo
spec:
 ...
 template:
   metadata:
     ...
     annotations:

>        prometheus.metrics.kuma.io/aggregate-my-service-enabled: "false"  # causes that configuration from Mesh is disabled and endpoint's metrics won't be exposed

Reviewer: ```suggestion
       prometheus.metrics.kuma.io/aggregate-my-service-enabled: "false"  # causes that configuration from Mesh to be disabled and result in this endpoint's metrics to not be exposed
```

       prometheus.metrics.kuma.io/aggregate-other-sidecar-port: "1234" # override port from Mesh
       prometheus.metrics.kuma.io/aggregate-application-port: "80"
       prometheus.metrics.kuma.io/aggregate-application-path: "/stats"
   spec:
     containers:
     ...
```
:::
::: tab "Universal"
```yaml
type: Dataplane
mesh: default
name: example
metrics:
 type: prometheus
 conf:
   path: /metrics/overridden
   aggregate:
     my-service: # causes that configuration from Mesh is disabled and endpoint's metrics won't be exposed
       enabled: false
     other-sidecar:
       port: 1234 # override port from Mesh

>      application:

Reviewer: We screwed up a little, we should have made this an array, that's the recommended way in CRDs... Array should be:

```yaml
- name: my-service
   enabled: false
   port: 1234
 ```

Author: My bad didn't know about recommended way. Should I create an issue that we should in the future change this to the list as a no-breaking change(new field)?
